prusti-viper/src/encoder/procedure_encoder.rs

```diff
@@ -2053,7 +2053,12 @@ impl<'p, 'v: 'p, 'tcx: 'v> ProcedureEncoder<'p, 'v, 'tcx> {
                                     )?);
                                 }
 
-                                _ => unreachable!()
+                                _ => {
+                                    unimplemented!(
+                                        "Only calls to closures are supported. The term at {:?} is a {:?}, not a closure.",
+                                        term.source_info.span, cl_type.kind()
+                                    )
+                                }
```

Comment:
Since you are at it, the even best fix would be to generate an internal encoding error. Something like:
```
cleanup(&self);
return Err(SpannedEncodingError::internal(
    format!("only calls to closures are supported. The term is a {:?}, not a closure.", cl_type.kind()),
    term.source_info.span,
));
```

The cleanup is sadly needed because of a hack (see https://github.com/viperproject/prusti-dev/issues/187).

The error message starts with a lowercase character just because it's the convention that `rustc` seems to use.